### PR TITLE
Guard the ready replica checking by server version

### PR DESCRIPTION
**Pull request by krousey**
I fixed replica readiness checking for 1.4->1.5 upgrades by using a field that only exists in versions >=1.4.0 in #36924 

This fixed a lot of issues in 1.4->1.5 upgrade testing, but did not fix 1.3->1.5 upgrade tests. I've disabled replica checking for 1.3 masters as the old logic was broken anyway.

This will not affect the 1.3 CI tests. Just 1.3 -> {1.4, 1.5} upgrade tests.

https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-gke-container_vm-1.3-container_vm-1.5-upgrade-cluster-new/330?log
is an example of this breakage. This is the tell-tale logs:

```console
Nov 22 09:40:50.469: INFO: 11 / 11 pods in namespace 'kube-system' are running and ready (506 seconds elapsed)
Nov 22 09:40:50.469: INFO: expected 5 pod replicas in namespace 'kube-system', 0 are Running and Ready.
Nov 22 09:40:50.469: INFO: POD  NODE  PHASE  GRACE  CONDITIONS
```

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/37303)

### CODE REVIEW

**Comment by krousey** [comment]
cc @saad-ali

**Comment by saad-ali** [comment]
> cc @saad-ali

@spxtr can you take a look this will help fix upgrade tests for 1.5

File: test/e2e/framework/util.go [hunk 1]
```diff
@@ -493,6 +495,14 @@ func WaitForPodsSuccess(c clientset.Interface, ns string, successPodLabels map[s
 // and some in Success. This is to allow the client to decide if "Success"
 // means "Ready" or not.
 func WaitForPodsRunningReady(c clientset.Interface, ns string, minPods int32, timeout time.Duration, ignoreLabels map[string]string) error {
+
+	// This can be removed when we no longer have 1.3 servers running with upgrade tests.
```
Comment (spxtr) [inline test/e2e/framework/util.go]: `TODO(krousey)`?

Comment (krousey) [inline test/e2e/framework/util.go]: https://github.com/kubernetes/kubernetes/issues/37310

**Comment by krousey** [comment]
@saad-ali: This should also be cherry-picked into 1.4 since the other one was.

**Comment by krousey** [comment]
@k8s-bot kubemark e2e test this

**Comment by krousey** [comment]
cc @roberthbailey

**Comment by saad-ali** [comment]
> @saad-ali: This should also be cherry-picked into 1.4 since the other one was.

Ack. Once this is merged to 1.5, we can slap a 1.4 milestone on there.

CC @jessfraz

**Comment by krousey** [comment]
@jessfraz did this ever make it into 1.4?

**Comment by jessfraz** [comment]
looks like it has not been cherry-picked yet, but we can get it in for the next release

**Comment by jessfraz** [comment]
cherry-picked in #37864

**Pull request merged**